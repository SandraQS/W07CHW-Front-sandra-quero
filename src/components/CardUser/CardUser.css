.section-user {
  background-color: black;
  margin-top: 35px;
  position: relative;
  border-radius: 10px;
  color: white;
  text-align: center;
}
.section-user__container-info {
  padding-top: 100px;
}

.section-user__container-image {
  width: 100px;
  height: 100px;
  border-radius: 50%;
  position: absolute;
  top: -20px;
}
.section-user__image {
  width: 100px;
  height: 100px;
  border: 2px solid black;
  border-radius: 50%;
  object-fit: cover;
}

.section-user__name {
  text-transform: uppercase;
  font-size: 25px;
}

.section-user__username {
  text-transform: uppercase;
  font-size: 15px;
}

button {
  margin-bottom: 5px;
}
