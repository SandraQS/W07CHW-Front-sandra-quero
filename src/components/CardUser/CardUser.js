import "./CardUser.css";

const CardUser = ({
  user: { image, name, username, age, friends, enemies, id },
}) => {
  return (
    <article
      className="section-user  col-md-6 col-lg-3 col-12 row justify-content-center"
      key={id}
    >
      <div className="section-user__container-image">
        <img src={image} alt={name} className="section-user__image" />
      </div>
      <div className="section-user__container-info row justify-content-between">
        <h2 className="section-user__name col-12">{name}</h2>
        <p className="section-user__username  col-md-6 col-12">{username}</p>
        <p className="section-user__age col-md-6 col-1">Edad: {age}</p>
        <p className="section-user__friends col-md-6 col-1">
          Amigos: {friends.length}
        </p>
        <p className="section-user__enemies col-md-6 col-1">
          Enemigos: {enemies.length}
        </p>
        <button
          type="button"
          className="section-user__button-friends btn btn-outline-light"
        >
          Añadir como amigo
        </button>
        <button
          type="button"
          className="section-user__button-enemies btn btn-outline-light"
        >
          Añadir como enemigo
        </button>
      </div>
    </article>
  );
};
export default CardUser;
