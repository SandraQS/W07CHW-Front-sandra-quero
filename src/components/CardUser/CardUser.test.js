import ReactTestRenderer from "react-test-renderer";
import CardUser from "./CardUser";

describe("Given", () => {
  describe("WHen", () => {
    test("Then it should", () => {
      const user = {
        name: "Carl",
        username: "Carlitus",
        password:
          "$2b$10$s3c31JSTdrwRMwW5jLaE6.ridoUF4lXQpSTQtECectp/aCmMNXswK",
        friends: [
          {
            $oid: "618ecc54008088cc372842c6",
          },
          {
            $oid: "618fa76a5b619d1c20dbc17c",
          },
        ],
        enemies: [],
        age: 33,
        image:
          "https://cdns.iconmonstr.com/wp-content/assets/preview/2012/240/iconmonstr-user-14.png",
      };
      const cardUser = ReactTestRenderer.create(
        <CardUser key={user.id} user={user} />
      );

      expect(cardUser.toJSON()).toMatchSnapshot();
    });
  });
});
