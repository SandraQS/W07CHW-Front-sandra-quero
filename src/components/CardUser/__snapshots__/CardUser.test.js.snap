// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given WHen Then it should 1`] = `
<article
  className="section-user  col-md-6 col-lg-3 col-12 row justify-content-center"
>
  <div
    className="section-user__container-image"
  >
    <img
      alt="Carl"
      className="section-user__image"
      src="https://cdns.iconmonstr.com/wp-content/assets/preview/2012/240/iconmonstr-user-14.png"
    />
  </div>
  <div
    className="section-user__container-info row justify-content-between"
  >
    <h2
      className="section-user__name col-12"
    >
      Carl
    </h2>
    <p
      className="section-user__username  col-md-6 col-12"
    >
      Carlitus
    </p>
    <p
      className="section-user__age col-md-6 col-1"
    >
      Edad: 
      33
    </p>
    <p
      className="section-user__friends col-md-6 col-1"
    >
      Amigos: 
      2
    </p>
    <p
      className="section-user__enemies col-md-6 col-1"
    >
      Enemigos: 
      0
    </p>
    <button
      className="section-user__button-friends btn btn-outline-light"
      type="button"
    >
      Añadir como amigo
    </button>
    <button
      className="section-user__button-enemies btn btn-outline-light"
      type="button"
    >
      Añadir como enemigo
    </button>
  </div>
</article>
`;
